sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/EventHubBufferedProducerClientTests.cs

```diff
@@ -1220,49 +1219,142 @@ public async Task FlushAsyncDrainsPartitions()
         /// </summary>
         ///
         [Test]
-        public async Task FlushAsyncWaitsForEventHandlersToFinish()
+        public async Task FlushAsyncInvokesTheSuccessHandlerAndWaitsForCompletion()
```

Comment:
This may already be covered in other tests, but if not should we add a test for the Success and Failure handler that throws to validate that it does not bubble up?

Reply:
It's a good thought - we can definitely validate that from the `Flush` perspective, since the publishing takes place in the foreground.   There's not really a good way to validate that from the publishing perspective, since its all done via ephemeral background tasks.